redshift_sqlalchemy/dialect.py

```diff
@@ -145,19 +187,91 @@ class RedshiftDialect(PGDialect_psycopg2):
         }),
         (schema.Table, {
             "ignore_search_path": False,
-            'diststyle': None,
-            'distkey': None,
-            'sortkey': None
+            "diststyle": None,
+            "distkey": None,
+            "sortkey": None,
+            "sortkey_type": COMPOUND_SORTKEY,
         }),
     ]
 
     @reflection.cache
+    def get_table_options(self, connection, table_name, schema, **kw):
+        insp = inspect(connection)
+        key = (schema or insp.default_schema_name, table_name)
+        tables, views = self._get_all_table_and_view_info(connection)
+        table_info = tables[key]
+        column_info = self._get_all_column_info(connection)
+        columns = column_info[key]
+        # If sortkey is interleaved, column numbers alternate negative values
+        keyfunc = lambda col: abs(int(col.sortkey))
```

Comment:
No lambdas please, PyPy can't optimize them as much as static functions

Certainly no assigned lambdas as they can always be replaced with an inner function `def`
```
def _get_table_options_keyfunc(col):
    return abs(int(col.sortkey))
```